### PR TITLE
build(tests): type check integration.test.ts

### LINKED ISSUE #352: build: enable full typescript checking of test files
# Problem

currently, test files are not checked by `npm run build`, we exclude them in tsconfig: https://github.com/neovim/node-client/blob/d3bc8d36b50ccc188e083b33ff537819e661dd88/packages/neovim/tsconfig.json#L7

# Expected behavior

- test files should be checked
- fix all the errors

### PR DESCRIPTION
**Pull request by gjf7**
Closes #352.

### CODE REVIEW

File: packages/integration-tests/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  testEnvironment: 'node',
+  testPathIgnorePatterns: [".d.ts", ".js"]
```
[the remark this answers is not in the document]
Comment (gjf7) [reply packages/integration-tests/jest.config.js:3]: How about moving the jest.config.js content into packages/integration-tests/package.json for consistency?

[the remark this answers is not in the document]
Comment (justinmk) [reply packages/integration-tests/jest.config.js]: Sure that's fine. I have the impression that the .config.js files are the preferred approach these days (at least for eslint), but we will probably remove jest eventually (https://github.com/neovim/node-client/issues/333) so doesn't matter for now :)

**Comment by gjf7** [comment]
The unreliable test: emits "disconnect" after quit failed again, can u help me to rerun this job ? @justinmk

File: packages/integration-tests/jest.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (gjf7) [reply packages/integration-tests/jest.config.js]: Great!After this is merged, I'll start working on it. Mocha seems like a good choices.

**Pull request merged**